File: Ridy-taxi-app/ios/driver/driver/Networking/HTTP/GetRegisterInfo.swift
//
//  GetRegisterInfo.swift
//  driver
//
//  Created by Manly Man on 11/23/19.
//  Copyright © 2019 minimal. All rights reserved.
//

import Foundation


class GetRegisterInfo: HTTPRequest {
    var path: String = "driver/get"
    typealias ResponseType = RegistrationInfo
    var params: [String : Any]?
    
    init(jwtToken: String) {
        self.params = [
            "token": jwtToken
        ]
    }
}

struct RegistrationInfo: Codable {
    var driver: Driver
    var services: [Service] = []
}

File: Ridy-taxi-app/android/driver/src/main/java/com/innomalist/taxi/driver/networking/http/GetRegisterInfo.kt
package com.innomalist.taxi.driver.networking.http

import com.innomalist.taxi.common.models.Driver
import com.innomalist.taxi.common.models.Service
import com.innomalist.taxi.common.networking.http.interfaces.HTTPRequest
import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

class GetRegisterInfo(jwtToken: String): HTTPRequest() {
    override val path: String = "driver/get"
    init {
        this.params = mapOf("token" to jwtToken)
    }
}
@JsonClass(generateAdapter = true)
data class RegistrationInfo(
    val driver: Driver,
    val services: List<Service>
)